server/src/main/java/io/spine/server/migration/mirror/MirrorMigration.java

```diff
@@ -0,0 +1,186 @@
+ *
+ * <p>{@code Mirror} projection was deprecated in Spine 2.x. Previously, it was used to store
+ * aggregates' states to allow their querying. A single projection stored states of <b>all</b>
+ * aggregates in a Bounded Context.
+ *
+ * <p>Now, {@linkplain EntityRecordStorage} is used to store the aggregate's states when it is
+ * open for querying. This storage is created on a <b>per-aggregate</b> basis. This enables
+ * storing of queryable state-based columns along the state itself. For this reason, the migration
+ * is also done on a per-aggregate basis.
+ *
+ * <p><b>An example usage</b>
+ *
+ * <p>1. Create an instance of `MirrorMigration`:
+ * <pre>
+ * var contextSpec = ContextSpec.singleTenant("...");
+ * var storageFactory = ServerEnvironment.instance().storageFactory();
+ * var migration = new MirrorMigration(contextSpec, storageFactory, ParcelAgg.class);
+ * </pre>
+ *
+ * <p>2. Create `MigrationSupervisor`. Here we are going to use a default implementation,
+ * which always continues the migration. Also, we will work with storages via batches
+ * of 500 records.
+ * <pre>
+ * var stepSize = 500;
+ * var supervisor = new MigrationSupervisor(stepSize);
+ * </pre>
+ *
+ * <p>3. Run the migration.
+ * <pre>
+ * migration.run(supervisor);
+ * </pre>
+ *
+ * @param <I>
+ *         the aggregate's identifier
+ * @param <S>
+ *         the aggregate's state
+ * @param <A>
+ *         the aggregate's class
+ */
+@Immutable
+public final class MirrorMigration<I, S extends EntityState<I>, A extends Aggregate<I, S, ?>> {
+
+    private final EntityRecordStorage<I, S> entityRecordStorage;
+    private final MirrorStorage mirrorStorage;
+    private final MirrorMapping<I, S, A> mapping;
+    private final String aggregateType;
+
+    /**
+     * Creates a new instance of {@code MirrorMigration}.
+     *
+     * <p>The created instance is able to migrate {@code Mirror} projections of the
+     * given aggregate class.
+     *
+     * @param contextSpec
+     *         the context to which the passed aggregate belongs
+     * @param factory
+     *         the storage provider
+     * @param aggClass
+     *         the class of an aggregate, {@code Mirror} projections of which are to be migrated
+     */
+    public MirrorMigration(ContextSpec contextSpec,
+                           StorageFactory factory,
+                           Class<A> aggClass) {
+
+        this.mirrorStorage = new MirrorStorage(contextSpec, factory);
+        this.entityRecordStorage = factory.createEntityRecordStorage(contextSpec, aggClass);
+        this.mapping = new MirrorMapping<>(aggClass);
+        this.aggregateType = AggregateClass.asAggregateClass(aggClass)
+                                          .stateTypeUrl()
+                                          .value();
+    }
+
+    /**
+     * Migrates {@linkplain Mirror} projections
+     * to the aggregate's {@linkplain EntityRecordStorage}.
+     */
+    public void run(MigrationSupervisor supervisor) {
```

Comment:
In addition to this method, I would expect an endpoint which takes the `MigrationStep`, e.g., `continueFrom(MigrationStep step, MigrationSupervisor supervisor)`.